site/content/en/docs/FAQ/delete_all.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "How to delete all tickets"
+linkTitle: "How to delete all tickets"
+weight: 3
+description: >
+  This guide provides guidance about how to delete all tickets in Open Match.
+---
+
+Open Match does not provides a `DeleteAll` API for users who want to delete all tickets in Open Match. However, starting from Open Match v0.10.0, we recommend using the `QueryTicketIds` API to query and delete all tickets. Here is a code snippet for the recommended usage.
+
+```go
+// Connect to QueryService.
+qc, err := grpc.Dial(queryServiceAddr, grpc.WithInsecure())
+if err != nil {
+    log.Fatalf("Failed to connect to Open Match, got %s", err.Error())
+}
+defer qc.Close()
+
+q := pb.NewQueryServiceClient(conn)
+
+fc, err := grpc.Dial(queryServiceAddr, grpc.WithInsecure())
+if err != nil {
+    log.Fatalf("Failed to connect to Open Match, got %s", err.Error())
+}
+defer fc.Close()
+
+f := pb.NewQueryServiceClient(conn)
+
+// an empty pool returns all ticket ids in Open Match
+stream, err := qc.QueryTicketIds(context.Background(), &pb.QueryTicketIdsRequest{Pool: nil})
+
+for {
+    resp, err := stream.Recv()
+    if err == io.EOF {
+        break
+    }
+    if err != nil {
+        // process the error
+    }
+
+    for _, id := range resp.GetIds() {
+        _, err = f.DeleteTicket(context.Background(), &pb.DeleteTicketRequest{TicketId: id})
+        if err != nil {
+            // process the error
+        }
+    }
+}
+```
```

Comment:
How does this handle pending tickets or tickets that are already assigned?

Reply:
So what is the agreement on #1136? I thought this doc was intended to document how to delete all tickets in OM?

Reply:
More than the documentation itself, I am concerned about the implementation for: 
https://github.com/googleforgames/open-match/pull/1151/

Looks like that goes through query service - so I assume you will only get Active IDs? So is this statement on that API proto accurate:
_If the Pool contains no Filters, QueryTicketIds will return all TicketIDs in the state storage._ ?

The primary user scenario here was to have an ability to cleanup Open Match state storage - and if there is a simpler way to do that, we should consider that. Providing a state specific API may be cleaner (get all pending / get all assigned / get all in all states) but thats a lower priority.

Also, we dont expect this to be called at scale - so going through query cache for this is not a requirement.